stripe/src/main/java/com/stripe/android/model/PaymentIntent.java

```diff
@@ -300,11 +370,65 @@ public JSONObject toJson() {
         map.put(FIELD_DESCRIPTION, mDescription);
         map.put(FIELD_LIVEMODE, mLiveMode);
         map.put(FIELD_NEXT_SOURCE_ACTION, mNextSourceAction);
+        map.put(FIELD_NEXT_ACTION, mNextAction);
         map.put(FIELD_RECEIPT_EMAIL, mReceiptEmail);
         map.put(FIELD_STATUS, mStatus);
         map.put(FIELD_SOURCE, mSource);
         StripeNetworkUtils.removeNullAndEmptyParams(map);
         return map;
     }
 
+    public enum NextActionType {
+        RedirectToUrl("redirect_to_url"),
+        AuthorizeWithUrl("authorize_with_url");
```

Comment:
I think if we can do the hard cut I suggested above, we should also be able to remove `authorize_with_url` since it is only shown in `next_source_action`, not `next_action`